DataManagementSystem/Agent/RequestOperations/RemoveReplica.py

```diff
@@ -132,29 +127,27 @@ def bulkRemoval( self, toRemoveDict, targetSE ):
 
     :param dict toRemoveDict: { lfn: opFile, ... }
     :param str targetSE: target SE name
-    :return: toRemoveDict with updated errors
     """
     removeReplicas = self.dm.removeReplica( targetSE, toRemoveDict.keys() )
     if not removeReplicas["OK"]:
-      for opFile in toRemoveDict.values():
+      for opFile in toRemoveDict.itervalues():
         opFile.Error = removeReplicas["Message"]
-      return S_ERROR( removeReplicas["Message"] )
+      return removeReplicas
     removeReplicas = removeReplicas["Value"]
     # # filter out failed
-    for lfn, opFile in toRemoveDict.items():
+    for lfn, opFile in toRemoveDict.iteritems():
       if lfn in removeReplicas["Failed"]:
         opFile.Error = str( removeReplicas["Failed"][lfn] )
-    return S_OK( toRemoveDict )
+    return S_OK()
 
-  def singleRemoval( self, opFile, targetSE ):
-    """ remove opFile replica from targetSE
+  def removeWithOwnerProxy( self, opFile, targetSE ):
```

Comment:
Right, indeed these methods should not be public!!! but this is the case of a large fraction of methods and 90% of data members of our classes... 
I can change that indeed! ;-)